### PR TITLE
feat(generator): detect paginated methods

### PR DESCRIPTION
**Pull request by codyoss**
As defined by AIP-4233

Updates: #297

### CODE REVIEW

File: generator/internal/api/model.go [hunk 1]
```diff
@@ -115,6 +115,8 @@ type Method struct {
 	OutputTypeID string
 	// PathInfo information about the HTTP request
 	PathInfo *PathInfo
+	// IsPagable is true if the method conforms to standard
```
Comment (coryan) [inline generator/internal/api/model.go:118]: s/IsPagable/IsPageable/ in the comment.

"to standard" ... ? Incomplete sentence? Maybe reference the AIP?

File: generator/internal/parser/testdata/pagination.proto [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+}
+
+message Foo {
+    option (google.api.resource) = {
+        type: "test.googleapis.com/Foo"
+        pattern: "projects/{project}/foos/{foo}"
+    };
+
+    string name = 1 [(google.api.field_behavior) = OUTPUT_ONLY];
+}
```
Comment (coryan) [inline generator/internal/parser/testdata/pagination.proto:127]: nit: newline at end of file.

**Review by coryan — COMMENTED** [review]
A couple of small nits. The test failures I know you will fix.

File: generator/internal/parser/testdata/pagination.proto (hunk 2 above)
Comment (codyoss) [reply generator/internal/parser/testdata/pagination.proto:127]: done.

File: generator/internal/api/model.go (hunk 1 above)
Comment (codyoss) [reply generator/internal/api/model.go:118]: done.

**Review by coryan — APPROVED** [review]
Ugh, I meant to approve before. Fixed now.

**Pull request merged**